pymatgen/io/cp2k/inputs.py

```diff
@@ -0,0 +1,1364 @@
+
+        super(Smear, self).__init__(
+            "SMEAR", description=description, subsections={}, keywords=keywords
+        )
+
+
+class BrokenSymmetry(Section):
+
+    """
+    Define the required atomic orbital occupation assigned in initialization
+    of the density matrix, by adding or subtracting electrons from specific
+    angular momentum channels. It works only with GUESS ATOMIC
+    """
+
+    def __init__(
+        self,
+        l_alpha=-1,
+        n_alpha=0,
+        nel_alpha=-1,
+        l_beta=-1,
+        n_beta=0,
+        nel_beta=-1,
+    ):
+        """
+        Initialize the broken symmetry section
+
+        Args:
+            l_alpha: Angular momentum quantum number of the orbitals whose occupation is changed
+            n_alpha: Principal quantum number of the orbitals whose occupation is changed.
+                Default is the first not occupied
+            nel_alpha: Orbital occupation change per angular momentum quantum number. In
+                unrestricted calculations applied to spin alpha
+            l_beta: Same as L_alpha for beta channel
+            n_beta: Same as N_alpha for beta channel
+            nel_beta: Same as NEL_alpha for beta channel
+        """
+
+        description = (
+            "Define the required atomic orbital occupation assigned in initialization "
+            + "of the density matrix, by adding or subtracting electrons from specific "
+            + "angular momentum channels. It works only with GUESS ATOMIC"
+        )
+
+        keywords_alpha = [
+            Keyword("L", l_alpha),
+            Keyword("N", n_alpha),
+            Keyword("NEL", nel_alpha),
+        ]
+        alpha = Section(
+            "ALPHA", keywords=keywords_alpha, subsections={}, repeats=False
+        )
+
+        keywords_beta = [
+            Keyword("L", l_beta),
+            Keyword("N", n_beta),
+            Keyword("NEL", nel_beta),
+        ]
+        beta = Section(
+            "BETA", keywords=keywords_beta, subsections={}, repeats=False
+        )
+
+        super(BrokenSymmetry, self).__init__(
+            "BS",
+            description=description,
+            subsections={"ALPHA": alpha, "BETA": beta},
+            keywords=[],
+            repeats=False,
+        )
+
+
+class XC_FUNCTIONAL(Section):
+
+    """
+    Defines the XC functional to use
+    """
+
+    def __init__(self, functional, subsections={}):
+        """
+        Initialize the XC_FUNCTIONAL class
+        """
+
+        location = "CP2K_INPUT/FORCE_EVAL/DFT/XC/XC_FUNCTIONAL"
+
+        built_in = [
+            "BL3YLP",
+            "BEEFVDW",
+            "BLYP",
+            "BP",
+            "HCTH120",
+            "LDA",
+            "NONE",
+            "NO_SHORTCUT",
+            "OLYP",
+            "PADE",
+            "PBE",
+            "PBE0",
+            "TPSS",
+        ]
+
+        if functional not in built_in:
+            raise Warning("The selected functional does not exist in CP2K.")
```

Comment:
Would it be possible to add support here for libxc functionals? Happy to discuss details via Slack.